libc-top-half/musl/src/env/__init_tls.c

```diff
@@ -16,6 +16,20 @@
 volatile int __thread_list_lock;
 
 #ifndef __wasilibc_unmodified_upstream
+
+extern unsigned char __heap_base;
+extern unsigned char __data_end;
+
+static inline void setup_default_stack_size()
+{
+	ptrdiff_t stack_size = &__heap_base - &__data_end;
```

Comment:
For this legacy path, we need to also take into account the case where somebody linked with `--stack-first`.   In this case the SP grows down address zero and the static data is after the stack region.  Maybe we can detect this case by compare SP to `__global_base`?

Reply:
Good point, will handle that case.